### PR TITLE
Permission selection on form build, other UI updates

### PR DESCRIPTION
**Pull request by loneil**
# Description
- Add settings for user type/idps on form build
- Other formatting stuff
- Navigate away from pages on form creation

Still needs some more proper error handling and success messaging.
An issue where if the non-creating user fills out the form it fails to populate on the readonly success screen.
Will address these in next PR but should get this structure reviewed and merged in soon.

## Types of changes

New feature (non-breaking change which adds functionality)

Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wxpbtr-tools.pathfinder.gov.bc.ca/dashboard?id=common-hosted-form-service-pr-21

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wxpbtr-tools.pathfinder.gov.bc.ca/dashboard?id=common-hosted-form-service-pr-21

File: app/frontend/src/components/designer/FormViewer.vue [hunk 1]
```diff
@@ -104,10 +119,15 @@ export default {
...
-            'Your form has been successfully submitted (UX here TBD)'
-          );
+          this.$router.push({
+            name: 'FormSubmissionView',
+            params: {
+              formId: this.formId,
+              versionId: response.data.formVersionId,
+              submissionId: response.data.id,
+            },
+            query: { success: true },
```
Comment (loneil) [inline app/frontend/src/components/designer/FormViewer.vue:129]: Likely not going this way in the future (don't think we can navigate to this page when a non-creating user fills out the form) so don't worry about this query param

File: app/frontend/src/filters/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import Vue from 'vue';
+import moment from 'moment';
```
Comment (loneil) [inline app/frontend/src/filters/index.js:2]: Just read an article about how this lib is getting deprecated (though will continue to be updated with timezone changes for the next few years), should keep in mind for all our apps for the future.
https://momentjs.com/docs/#/-project-status/

**Review by TimCsaky — APPROVED** [review]

File: app/frontend/src/components/designer/FormDesigner.vue [hunk 3]
```diff
@@ -1,29 +1,85 @@
...
+                />
+              </v-col>
+              <v-col cols="12" xl="8">
+                <v-text-field
+                  dense
+                  flat
+                  label="Description"
+                  data-test="text-formDescription"
+                  v-model="formDescription"
+                  :rules="formDescriptionRules"
```
Comment (parc-jason) [inline app/frontend/src/components/designer/FormDesigner.vue:28]: Description isn't mandatory in the db, should it be?

File: app/frontend/src/components/designer/FormDesigner.vue [hunk 4]
```diff
@@ -1,29 +1,85 @@
...
+            </v-row>
+            <p>Select which type of user can fill out out this form once published</p>
+            <v-radio-group v-model="userType" :mandatory="false" :rules="loginRequiredRules">
+              <v-radio disabled label="Public (annonymous)" :value="ID_PROVIDERS.PUBLIC"></v-radio>
+              <v-radio label="Log-in Required" value="login"></v-radio>
+              <div v-if="userType === 'login'" class="pl-5 mb-5">
+                <v-row>
+                  <v-checkbox v-model="idps" class="mx-4" label="IDIR" :value="ID_PROVIDERS.IDIR"></v-checkbox>
+                  <v-checkbox
+                    disabled
```
Comment (parc-jason) [inline app/frontend/src/components/designer/FormDesigner.vue:40]: Just a comment, we should start the paperwork for other IDPs...

File: app/frontend/src/filters/index.js (hunk 2 above)
Comment (parc-jason) [reply app/frontend/src/filters/index.js:2]: uh oh...

**Review by parc-jason — APPROVED** [review]
good stuff.

File: app/frontend/src/components/designer/FormViewer.vue (hunk 1 above)
Comment (loneil) [reply app/frontend/src/components/designer/FormViewer.vue:129]: This is going away next pr

File: app/frontend/src/components/designer/FormDesigner.vue (hunk 3 above)
Comment (loneil) [reply app/frontend/src/components/designer/FormDesigner.vue:28]: It was failing to POST with a 422 without it, just made it mandatory for demo

Comment (parc-jason) [reply app/frontend/src/components/designer/FormDesigner.vue:28]: ahh, it's the schema validation rule... 
models , line 79 should have ```{ type: ['string', 'null'], maxLength: 255 }``` to allow nulls or empty strings and enforce the length limit.

Comment (loneil) [reply app/frontend/src/components/designer/FormDesigner.vue:28]: Ok thanks, will add that in my next PR and make the field optional on the FE too

**Review by jujaga — APPROVED** [review]

**Pull request merged**